src/com/mendix/widget/slider/Slider.ts

```diff
@@ -0,0 +1,166 @@
+    upperBoundAttribute: string;
+    maxAttribute: string;
+    minAttribute: string;
+    onClickMicroflow: string;
+    stepValue: number;
+    stepAttribute: string;
+    noOfMarkers: number;
+    tooltipText: string;
+    showRange: boolean;
+    decimalPlaces: number;
+
+    private contextObject: mendix.lib.MxObject;
+
+    postCreate() {
+        this.onClick = this.onClick.bind(this);
+        this.onChange = this.onChange.bind(this);
+    }
+
+    update(contextObject: mendix.lib.MxObject, callback?: Function) {
+        this.contextObject = contextObject;
+        this.resetSubscriptions();
+        this.updateRendering();
+
+        if (callback) {
+            callback();
+        }
+    }
+
+    uninitialize(): boolean {
+        unmountComponentAtNode(this.domNode);
+
+        return true;
+    }
+
+    private updateRendering(validationMessage?: string) {
+        const disabled = !this.contextObject
+            || this.readOnly
+            || !!(this.stepAttribute && this.contextObject.isReadonlyAttr(this.stepAttribute));
+        render(createElement(SliderComponent, {
+            decimalPlaces: this.decimalPlaces,
+            disabled,
+            hasError: !!validationMessage,
+            lowerBound: this.getAttributeValue(this.lowerBoundAttribute),
+            maxValue: this.getAttributeValue(this.maxAttribute),
+            minValue: this.getAttributeValue(this.minAttribute),
+            noOfMarkers: this.noOfMarkers,
+            onChange: this.onChange,
+            onClick : this.onClick,
+            showRange: this.showRange,
+            stepValue: this.getAttributeValue(this.stepAttribute, this.stepValue),
+            tooltipText: this.tooltipText,
+            upperBound: this.getAttributeValue(this.upperBoundAttribute),
+            value: this.getAttributeValue(this.valueAttribute),
+            validationMessage
+        }), this.domNode);
+    }
+
+    private getAttributeValue(attributeName: string, defaultValue?: number): number | undefined {
+        if (this.contextObject && attributeName) {
+            if (this.contextObject.get(attributeName) !== "") {
+                return parseFloat(this.contextObject.get(attributeName) as string);
+            }
+        }
+
+        return defaultValue;
+    }
+
+    private handleValidation(validations: mendix.lib.ObjectValidation[]) {
+        const validationMessage = validations[0].getErrorReason(this.valueAttribute);
+        if (validationMessage) {
+            this.updateRendering(validationMessage);
+        }
+    }
+
+    private onChange(value: number) {
+        if ((value || value === 0) && !this.showRange) {
+            if (value > this.getAttributeValue(this.maxAttribute)) {
+                this.contextObject.set(this.valueAttribute, this.getAttributeValue(this.maxAttribute));
+            } else {
+                this.contextObject.set(this.valueAttribute, value);
+            }
+        } else if (Array.isArray(value)) {
```

Comment:
The onChange takes in a number array when its a range interval. so the input so be a number or a number array